src/ops.rs

```diff
@@ -0,0 +1,411 @@
+use crate::array::ArrayElement;
+use crate::{array::Array, stream::StreamOrDevice};
+use num_traits::NumCast;
+
+impl Array {
+    /// Construct an array of zeros.
+    ///
+    /// Example:
+    ///
+    /// ```rust
+    /// use mlx::{array::Array, stream::StreamOrDevice};
+    /// Array::zeros::<f32>(&[5, 10], StreamOrDevice::default());
+    /// ```
+    ///
+    /// # Parameters:
+    /// - shape: Desired shape
+    /// - stream: Stream or device to evaluate on
+    pub fn zeros<T: ArrayElement>(shape: &[usize], stream: StreamOrDevice) -> Array {
+        let shape = shape.iter().map(|x| *x as i32).collect::<Vec<i32>>();
```

Comment:
We should check for overflow when converting from `usize` to `i32`

Reply:
Or just make it `&[i32]` to avoid additional memory allocation

Reply:
Did that! Makes more sense